src/Microsoft.SqlTools.SqlCore/TableDesigner/TableDesignerManager.cs

```diff
@@ -1822,6 +1825,32 @@ private void CheckCreateAlterTablePermission(TableInfo tableInfo, string connect
                 });
             }
         }
+
+        private SqlConnectionStringBuilder GetConnectionStringBuilderForTable(TableInfo tableInfo)
+        {
+            var connectionStringBuilder = new SqlConnectionStringBuilder(tableInfo.ConnectionString);
+            connectionStringBuilder.InitialCatalog = tableInfo.Database;
+            connectionStringBuilder.ApplicationName = TableDesignerManager.TableDesignerApplicationNameSuffix;
+            return connectionStringBuilder;
+        }
+
+        private List<string> GetSchemaNames(TableInfo tableInfo, string connectionString)
+        {
+            var schemaNames = new List<string>();
```

Comment:
looks fine, and I guess we don't have a SMO object model here to use that for getting Schemas, so this looks quicker.